docs/reference/api.md

```diff
@@ -0,0 +1,214 @@
+- Body: data returned by Pebble server in JSON format.
+
+### JSON format
+
+Pebble API uses JSON for data exchange. Both request and response bodies are formatted as JSON objects or arrays.
+
+Example JSON Response:
+
+```json
+{
+  "type": "sync",
+  "status-code": 200,
+  "status": "OK",
+  "result": {
+    "name": "svc1",
+    "startup": "disabled",
+    "current": "stopped"
+  }
+}
+```
+
+### Request types
+
+Pebble API supports both synchronous and asynchronous requests.
+
+- Synchronous requests: The client sends a request and waits for the server to return a response before continuing.
+- Asynchronous requests: For long-running operations, Pebble API offers asynchronous processing. The client sends a request and gets an acknowledgment immediately, but the actual processing is still running on the server side. The client can poll for the result. Specific endpoints supporting asynchronous operations are documented below with `type` as `async`.
+
+## Using cURL
+
+To access API endpoints over the Unix socket, use the `--unix-socket` option of `curl`. For example:
+
+* TODO
+
+    ```
+    curl ...
+    ```
+
+* TODO
+
+    ```
+    curl ...
+    ```
+
+## Using the Go client
+
+To use the Go client to access API endpoints over the Unix socket, first create a client using `New`, and then call the methods on the returned Client to interact with the API. Example:
+
+```go
+import (
+  "log"
+
+	"github.com/canonical/pebble/client"
+)
+
+func main() {
+	pebble, err := client.New(&client.Config{Socket: ".pebble.socket"})
+	if err != nil {
+		log.Fatal(err)
+	}
+	changeID, err := pebble.Stop(&client.ServiceOptions{Names: []string{"mysvc"}})
+	if err != nil {
+		log.Fatal(err)
+	}
+	_, err = pebble.WaitChange(changeID, &client.WaitChangeOptions{})
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+```
+
+We try to never change the underlying HTTP API in a backwards-incompatible way. However, in rare cases we may change the Go client in a backwards-incompatible way.
+
+For more information, see the [Go client documentation](https://pkg.go.dev/github.com/canonical/pebble/client).
+
+## Using the Python client
+
+The Ops library for writing and testing Juju charms includes a [Python client for the Pebble API](https://ops.readthedocs.io/en/latest/reference/pebble.html). You can use the Python client to access API endpoints over the Unix socket. For example:
+
+```python
+import ops
+client = ops.pebble.Client('/path/to/.pebble.socket')
+client.start_services(['svc1', 'svc2'])
+```
+
+For more information, see:
+
+- [Ops library documentation for the Python client](https://ops.readthedocs.io/en/latest/reference/pebble.html#ops.pebble.Client)
+- [Source code of the Python client](https://github.com/canonical/operator/blob/main/ops/pebble.py)
+- [Pebble in the context of Juju charms](https://juju.is/docs/sdk/interact-with-pebble)
+
+For more examples, see "How to use the Pebble API". <!-- [David] Link to the how-to guide -->
```

Comment:
Will we have Python examples in the how-to doc? Maybe we could put this link at the end of the curl section and the Go section instead

Reply:
For completeness, let's include Python examples in the how-to doc.

Here I didn't mean "for more Python examples (since this is the end of the Python section), see ...", but rather:

```
## Accessing the API
### curl
### go
### python
### more examples - link to how-to
```

But since we removed the `Accessing the API` subsection because of previous comments, now the link is in an awkward position. If we put it under curl, it means "for more curl examples".

How about:

```
## Accessing Pebble API
Here are some methods to access the API. For more examples, see ...
### curl
### go
### python
```